wdl-format/src/format.rs

```diff
@@ -0,0 +1,615 @@
+//! A module for formatting WDL code.
+
+use std::fmt::Write;
+
+use anyhow::Result;
+use format_state::SPACE;
+use wdl_ast::v1::Decl;
+use wdl_ast::v1::DocumentItem;
+use wdl_ast::v1::Expr;
+use wdl_ast::v1::HintsItem;
+use wdl_ast::v1::HintsSection;
+use wdl_ast::v1::InputSection;
+use wdl_ast::v1::LiteralBoolean;
+use wdl_ast::v1::LiteralFloat;
+use wdl_ast::v1::LiteralInteger;
+use wdl_ast::v1::LiteralString;
+use wdl_ast::v1::OutputSection;
+use wdl_ast::v1::StringPart;
+use wdl_ast::v1::StructDefinition;
+use wdl_ast::v1::Type;
+use wdl_ast::AstNode;
+use wdl_ast::AstToken;
+use wdl_ast::Comment;
+use wdl_ast::Diagnostic;
+use wdl_ast::Direction;
+use wdl_ast::Document;
+use wdl_ast::Ident;
+use wdl_ast::SyntaxElement;
+use wdl_ast::SyntaxKind;
+use wdl_ast::Validator;
+use wdl_ast::Version;
+use wdl_ast::VersionStatement;
+
+mod comments;
+mod format_state;
+mod import;
+mod metadata;
+mod task;
+mod workflow;
+
+use comments::format_inline_comment;
+use comments::format_preceding_comments;
+use format_state::FormatState;
```

Comment:
Nit: I generally don't like excessive duplication in the naming of things. What we've got here (even though it's private) _would be_ a `wdl_format::format::format_state::FormatState` which seems—like many duplications of the word "format".

My feeling is it can just be `wdl_format::state::State`:

- You only export `format` within `lib.rs`, and the crate is named `wdl-format`, so I don't feel that layer is needed at all (would suggest just moving everything up a level and making `format.rs` as `lib.rs`).
- `format_state` could be renamed as simply `State`, as it's implied that it's the state related to formatting given that it's under the `format` module.

FWIW, generally the upside of more verbosely naming things is to avoid name duplication and/or confusion in modules where multiple ideas conflict. In these cases though, you can just `use format::State as FormatState`.